crates/trie/trie/src/node_iter.rs

```diff
@@ -126,6 +126,20 @@ where
     ///
     /// If `metrics` feature is enabled, also updates the metrics.
     fn seek_hashed_entry(&mut self, key: B256) -> Result<Option<(B256, H::Value)>, DatabaseError> {
+        if let Some((last_key, last_value)) = self.last_next_result {
+            if last_key == key {
+                trace!(target: "trie::node_iter", seek_key = ?key, "reusing result from last next() call instead of seeking");
+                self.last_next_result = None; // Consume the cached value
+
+                let result = Some((last_key, last_value));
+                self.last_seeked_hashed_entry = Some(SeekedHashedEntry { seeked_key: key, result });
+
+                return Ok(result);
+            }
+        }
+
+        self.last_next_result = None;
```

Comment:
Do we need to set these to `None` here? If so could we also add a comment as to why? I don't expect we would be seeking the same entry multiple times in a row, so this seems fine. However I also don't think we necessarily need to re-set, the only upside would be that we might do fewer `last_seek == key` comparisons when we are doing many seeks for different keys in a row.

Reply:
Yeah, I added this initially to prevent a potentially stale result from an earlier `next()` call if the `TrieWalker` skipped ahead significantly. Realised it's not needed there after your comment. Removed now!